### PR TITLE
Docs: Update data flow documentation to the new API.

### PR DESCRIPTION
**Pull request by aschackmull**
This updates the pieces of documentation that I could find to the new data flow API.
I identified the places that needed updates, by grepping for the following list of words:
```
::Configuration
isSanitizer
AdditionalTaintStep
hasFlow
isBarrierGuard
DataFlow::PathNode
DataFlow::PathGraph
```

### CODE REVIEW

**Comment by mattpollard** [comment]
👋🏻 @felicitymay, I'll remove this from the Docs Content FR project as it's now in y'all's project for triage. Let me know if I should handle this differently.

File: docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst [hunk 1]
```diff
@@ -168,74 +168,61 @@ Global data flow tracks data flow throughout the entire program, and is therefor
...
--  ``isSink``—defines where taint may flow to
--  ``isSanitizer``—optional, restricts the taint flow
--  ``isSanitizerGuard``—optional, restricts the taint flow
--  ``isAdditionalTaintStep``—optional, adds additional taint steps
+   module MyFlow = TaintTracking::Global<MyFlowConfiguration>;
 
-Similar to global data flow, the characteristic predicate ``MyTaintTrackingConfiguration()`` defines the unique name of the configuration, so ``"MyTaintTrackingConfiguration"`` should be replaced by the name of your class.
-
-The taint tracking analysis is performed using the predicate ``hasFlow(DataFlow::Node source, DataFlow::Node sink)``.
+The resulting module is completely similar to the one obtained from ``DataFlow::Global<ConfigSig>``.
```
Comment (subatoi) [inline docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst:225]: Is there an alternative way of expressing "completely similar", if that's not a technical term I'm unfamiliar with? Does this mean "exactly the same as" for example?

(There are six instances of "completely similar")

**Review by subatoi — DISMISSED** [review]
Just had one comment/question from Docs, thank you!

File: docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst (hunk 1 above)
Comment (subatoi) [reply docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst:225]: Thanks for the explanation! How about(?):

```suggestion
The resulting module is equivalent to the one obtained from ``DataFlow::Global<ConfigSig>``.
```

I think this should imply that the modules are the same in the context in which they're intended to be used, but that the modules themselves are not necessarily identical.

Comment (aschackmull) [reply docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst:225]: Hmm, perhaps this?
```
The resulting module has an identical signature to the one obtained from ``DataFlow::Global<ConfigSig>``.
```

Comment (subatoi) [reply docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst:225]: That sounds good!

**Review by geoffw0 — DISMISSED** [review]
LGTM (focussed on CPP, and I didn't check the examples compile).

Good to know we have partial flow support in new data flow.

File: docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst (hunk 1 above)
Comment (rvermeulen) [reply docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst:225]: How about identical interface?

Comment (aschackmull) [reply docs/codeql/codeql-language-guides/analyzing-data-flow-in-cpp-new.rst:225]: That could work as well, although I prefer "signature" as that's the proper technical term (QL _modules_ implement _signatures_).

**Review by subatoi — APPROVED** [review]

**Comment by jketema** [comment]
One high-level comment: internally we standardised the names of our configurations to end in `Config` should we do the same here?

**Comment by aschackmull** [comment]
> One high-level comment: internally we standardised the names of our configurations to end in `Config` should we do the same here?

I have no strong opinion on this. Feel free to make a follow-up PR if you like.

**Pull request merged**